x-pack/legacy/plugins/apm/public/components/shared/Links/apm/APMLink.tsx

```diff
@@ -18,6 +18,8 @@ interface Props extends EuiLinkAnchorProps {
   children?: React.ReactNode;
 }
 
+export type APMLinkExtendProps = Omit<Props, 'path' | 'query'>;
+
```

Comment:
Not necessary for this PR but looking at the TODO in this file we could fix it like:

```ts
export function getAPMHref(
  path: string,
  urlParams: IUrlParams,
  query: APMQueryParams = {}
) {
  const nextQuery = {
    rangeFrom: urlParams.rangeFrom || TIMEPICKER_DEFAULTS.rangeFrom,
    rangeTo: urlParams.rangeTo || TIMEPICKER_DEFAULTS.rangeTo,
    refreshPaused: urlParams.refreshPaused || TIMEPICKER_DEFAULTS.refreshPaused,
    refreshInterval:
      urlParams.refreshInterval || TIMEPICKER_DEFAULTS.refreshInterval,
    kuery: urlParams.kuery,
    environment: urlParams.environment,
    ...query
  };
  const nextSearch = fromQuery(nextQuery);

  return url.format({
    pathname: '',
    hash: `${path}?${nextSearch}`
  });
}
```

I opted for inlining `PERSISTENT_APM_PARAMS` since it's not re-used anywhere else and this seemed a little clearer. 
But again: not important for this PR.